mne/io/cnt/cnt.py

```diff
@@ -599,3 +786,238 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+            or (events is not None and event_id is not None)
+        ):
+            raise ValueError("Both `events` and `event_id` must be " "None or not None")
+
+        assert data.shape == (
+            self._raw_extras[0]["n_epochs"],
+            self.info["nchan"],
+            self._raw_extras[0]["n_pnts"],
+        )
+        tmax = ((data.shape[2] - 1) / self.info["sfreq"]) + tmin
+
+        super().__init__(
+            self.info,
+            data,
+            events,
+            event_id,
+            tmin,
+            tmax,
+            baseline,
+            reject=reject,
+            flat=flat,
+            reject_tmin=reject_tmin,
+            reject_tmax=reject_tmax,
+            filename=input_fname,
+            verbose=verbose,
+        )
+
+        # data are preloaded but _bad_dropped is not set so we do it here:
+        self._bad_dropped = True
+
+        logger.info("Ready.")
+
+    def _read_cnt_epochs_data(self):
+        """Read epochs data from .eeg file."""
+        cnt_info = self._raw_extras[0]
+        info = self.info
+
+        input_fname = cnt_info["input_fname"]
+
+        n_epochs = cnt_info["n_epochs"]
+        # number of points per epoch
+        n_pnts = cnt_info["n_pnts"]
+        n_channels = cnt_info["orig_nchan"]
+        cals = [d["cal"] for d in info["chs"]]
+
+        data = np.empty((n_epochs, n_channels, n_pnts), dtype=float)
+        epoch_headers = []
+        SWEEP_HEAD_SIZE = 13
+        DATA_POINT_SIZE = 4
+
+        with open(input_fname, "rb") as f:
+            # Ensure the file pointer is at the beginning of the EEG data
+
+            data_start = 900 + n_channels * 75
+            data_end = data_start + n_epochs * (
+                SWEEP_HEAD_SIZE + n_pnts * n_channels * DATA_POINT_SIZE
+            )
+            data_step = SWEEP_HEAD_SIZE + n_pnts * n_channels * DATA_POINT_SIZE
+
+            for epoch, i in enumerate(range(data_start, data_end, data_step)):
+                # Epoch Header:
+                # char Accept; /* accept byte */
+                # short Ttype; /* trial type */
+                # short Correct; /* accuracy */
+                # float Rt; /* reaction time */
+                # short Response; /* response type */
+                # short Reserved; /* not used */
+                f.seek(i)
+                epoch_header = np.fromfile(
+                    f,
+                    dtype=np.dtype("<c,h,h,f,h,h"),
+                    count=1,
+                )
+                print(epoch_header)
+                # last short not used
+                epoch_headers.append(epoch_header[0])
+                f.seek(i + SWEEP_HEAD_SIZE)
+                # Read the data points as a 4-byte integer array
+                data_points = np.fromfile(
+                    f,
+                    dtype=np.dtype("<i4,"),  # little-endian 4-byte integer
+                    count=n_pnts * n_channels,
+                )
```

Comment:
@larsoner Feel free to ignore if you don't have the time to address this, but when I was testing reading the actual blocks of data, only first and last data points were being read properly. Here is the correct data read using the struct library:
```
for point in range(pnts):
    for channel in range(chan):

          # Read the data point as a 4-byte integer
          value = struct.unpack('<l', f.read(DATA_POINT_SIZE))[0]

            # Scale the data point to microvolts and store it in the data array
            data[sweep, channel, point] = value * factors[channel]
```
Output:
```
array([[-8.70227814e-06,  5.48362732e-06,  1.68085098e-05, ...,
        -1.91330910e-05, -2.33352184e-05, -2.75671482e-05],
       [-5.84125519e-06, -3.03983688e-06, -1.49011612e-07, ...,
        -1.35898590e-05, -1.48415565e-05, -1.49905682e-05],
       [-1.23381615e-05, -4.94718552e-06, -2.17556953e-06, ...,
        -6.28829002e-06, -7.86781311e-06, -6.91413879e-06],
       ...,
       [-7.36117363e-06, -8.94069672e-08,  3.75509262e-06, ...,
        -1.31428242e-05, -1.14440918e-05, -1.03712082e-05],
       [ 9.47713852e-06,  9.23871994e-06,  9.00030136e-06, ...,
         2.47359276e-05,  2.49445438e-05,  2.52127647e-05],
       [ 4.88758087e-06,  9.65595245e-06,  8.34465027e-06, ...,
        -2.65836716e-05, -2.85208225e-05, -2.87592411e-05]])
```
Current implementation output:
```
array([[-8.70227814e-06, -5.84125519e-06, -1.23381615e-05, ...,
        -6.85453415e-06, -7.68899918e-06, -7.12275505e-06],
       [-4.73856926e-06, -9.03010368e-06,  2.45869160e-05, ...,
         1.27851963e-05,  1.09374523e-05,  1.60932541e-05],
       [ 1.51097775e-05,  1.63316727e-05,  1.10864639e-05, ...,
        -2.68220901e-06, -3.90410423e-06,  4.02331352e-06],
       ...,
       [-1.50501728e-05, -8.97049904e-06, -1.77323818e-05, ...,
        -2.05934048e-05,  1.44541264e-05, -1.72555447e-05],
       [-5.42402267e-06, -1.67191029e-05, -4.88758087e-06, ...,
        -1.73747540e-05, -2.08914280e-05, -1.72853470e-05],
       [-2.11298466e-05, -1.82092190e-05, -2.18451023e-05, ...,
        -1.03712082e-05,  2.52127647e-05, -2.87592411e-05]])
```

Do you have any idea why this would be the case? I may just switch to using something else if I cannot figure out the reason.